From 21722d9abad6488a5a34ff579c22d83014bca54f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ale=C5=A1=20Mat=C4=9Bj?= <amatej@redhat.com>
Date: Thu, 15 Jan 2026 12:22:09 +0100
Subject: [PATCH 8/8] Replace all `std::format` with `fmt::format`
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Otherwise it produces errors like:
```
In file included from /usr/include/c++/16/bits/stl_algobase.h:66,
                 from /usr/include/c++/16/bits/specfun.h:44,
                 from /usr/include/c++/16/cmath:3913,
                 from /usr/include/fmt/format.h:44,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/utils/format.hpp:23,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/common/exception.hpp:25,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/base/transaction_errors.hpp:24,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/base/transaction.hpp:24,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5-cli/exception.hpp:27,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5-cli/argument_parser_errors.hpp:23,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5-cli/argument_parser.hpp:23,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/dnf5/include/dnf5/context.hpp:26,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/dnf5-plugins/manifest_plugin/manifest.hpp:7,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/dnf5-plugins/manifest_plugin/manifest_resolve.cpp:4:
In member function ‘__gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>::operator++(int)’,
    inlined from ‘std::__unicode::_Utf_iterator<char32_t, char, __gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>, __gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>, std::__unicode::_Repl>::_M_read_utf32()’ at /usr/include/c++/16/bits/unicode.h:440:27,
    inlined from ‘std::__unicode::_Utf_iterator<char32_t, char, __gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>, __gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>, std::__unicode::_Repl>::_M_read()’ at /usr/include/c++/16/bits/unicode.h:272:19,
    inlined from ‘_ZNSt9__unicode13_Utf_iteratorIDicN9__gnu_cxx17__normal_iteratorIPKDiNSt4spanIS3_Lm1EE10__iter_tagEEES8_NS_5_ReplEEC4ES8_S8_S8_Q22bidirectional_iteratorIT1_E’ at /usr/include/c++/16/bits/unicode.h:132:11,
    inlined from ‘std::__unicode::_Utf_view<char, std::span<char32_t const, 1ul> >::_M_end<__gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>, __gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag> >(__gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>, __gnu_cxx::__normal_iterator<char32_t const*, std::span<char32_t const, 1ul>::__iter_tag>)auto’ at /usr/include/c++/16/bits/unicode.h:724:46,
    inlined from ‘std::__unicode::_Utf_view<char, std::span<char32_t const, 1ul> >::end()’ at /usr/include/c++/16/bits/unicode.h:739:22,
    inlined from ‘std::__format::__write_padded<std::__format::_Sink_iter<char>, char>(std::__format::_Sink_iter<char>, std::basic_string_view<char, std::char_traits<char> >, std::__format::_Align, unsigned long, char32_t)std::__format::_Sink_iter<char>’ at /usr/include/c++/16/format:860:27:
/usr/include/c++/16/bits/stl_iterator.h:1111:16: error: array subscript 2 is outside array bounds of ‘const char32_t[1]’ [-Werror=array-bounds=]
 1111 |       { return __normal_iterator(_M_current++); }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/16/bits/chrono_io.h:40,
                 from /usr/include/c++/16/chrono:3679,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/logger/logger.hpp:28,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/conf/config.hpp:29,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/conf/config_main.hpp:23,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/base/goal_elements.hpp:26,
                 from /builddir/build/BUILD/dnf5-5.4.0.0-build/dnf5-git-4298.63dd542/include/libdnf5/base/transaction.hpp:27:
/usr/include/c++/16/format: In function ‘std::__format::__write_padded<std::__format::_Sink_iter<char>, char>(std::__format::_Sink_iter<char>, std::basic_string_view<char, std::char_traits<char> >, std::__format::_Align, unsigned long, char32_t)std::__format::_Sink_iter<char>’:
/usr/include/c++/16/format:858:28: note: at offset 8 into object ‘__arr’ of size 4
  858 |             const char32_t __arr[1]{ __fill_char };
      |                            ^~~~~
```
---
 dnf5-plugins/manifest_plugin/manifest.cpp         | 2 +-
 dnf5-plugins/manifest_plugin/manifest_new.cpp     | 2 +-
 dnf5-plugins/manifest_plugin/manifest_resolve.cpp | 2 +-
 libdnf5/conf/config_main.cpp                      | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/dnf5-plugins/manifest_plugin/manifest.cpp b/dnf5-plugins/manifest_plugin/manifest.cpp
index d2852776..0ff92375 100644
--- a/dnf5-plugins/manifest_plugin/manifest.cpp
+++ b/dnf5-plugins/manifest_plugin/manifest.cpp
@@ -92,7 +92,7 @@ std::filesystem::path get_manifest_path(libdnf5::OptionPath & option, const std:
     std::string path{option.get_value()};
 
     if (option.get_priority() <= libdnf5::Option::Priority::DEFAULT) {
-        const auto & arch_path = std::regex_replace(path, std::regex("\\.yaml$"), std::format(".{}.yaml", arch));
+        const auto & arch_path = std::regex_replace(path, std::regex("\\.yaml$"), fmt::format(".{}.yaml", arch));
         const bool path_exists = std::filesystem::exists(path);
         const bool arch_path_exists = std::filesystem::exists(arch_path);
         if (path_exists && arch_path_exists) {
diff --git a/dnf5-plugins/manifest_plugin/manifest_new.cpp b/dnf5-plugins/manifest_plugin/manifest_new.cpp
index 18d75b12..2dc5aafe 100644
--- a/dnf5-plugins/manifest_plugin/manifest_new.cpp
+++ b/dnf5-plugins/manifest_plugin/manifest_new.cpp
@@ -133,7 +133,7 @@ void ManifestNewCommand::run() {
             populate_manifest(manifest, arch, false);
 
             std::string path{manifest_path_option->get_value()};
-            path = std::regex_replace(path, std::regex("\\.yaml$"), std::format(".{}.yaml", arch));
+            path = std::regex_replace(path, std::regex("\\.yaml$"), fmt::format(".{}.yaml", arch));
             serializer.serialize(manifest, path);
         }
     } else {
diff --git a/dnf5-plugins/manifest_plugin/manifest_resolve.cpp b/dnf5-plugins/manifest_plugin/manifest_resolve.cpp
index e08030b0..1b8c1a65 100644
--- a/dnf5-plugins/manifest_plugin/manifest_resolve.cpp
+++ b/dnf5-plugins/manifest_plugin/manifest_resolve.cpp
@@ -129,7 +129,7 @@ void ManifestResolveCommand::run() {
             populate_manifest(manifest, arch, false);
 
             std::string path{manifest_path_option->get_value()};
-            path = std::regex_replace(path, std::regex("\\.yaml$"), std::format(".{}.yaml", arch));
+            path = std::regex_replace(path, std::regex("\\.yaml$"), fmt::format(".{}.yaml", arch));
             serializer.serialize(manifest, path);
         }
     } else {
diff --git a/libdnf5/conf/config_main.cpp b/libdnf5/conf/config_main.cpp
index 9f284e35..77272618 100644
--- a/libdnf5/conf/config_main.cpp
+++ b/libdnf5/conf/config_main.cpp
@@ -97,7 +97,7 @@ static std::string get_user_agent() {
         return "libdnf";
     }
 
-    return std::format(
+    return fmt::format(
         "libdnf ({} {}; {}; {}.{})",
         os_release.get_value("NAME"),
         os_release.get_value("VERSION_ID"),
-- 
2.52.0

